eth/rlp/transactions.py

```diff
@@ -92,7 +94,9 @@ def sender(self) -> Address:
         """
         Convenience property for the return value of `get_sender`
         """
-        return self.get_sender()
+        if self._sender_cache is None:
```

Comment:
Can we just go ahead and steal [`cached_property`](https://github.com/django/django/blob/7e6b214ed34f5562dbd83cf54924a5b589a29715/django/utils/functional.py#L16-L58) from django....  It's actually better than this since future lookups pull straight from `__dict__` and bypass the descriptor entirely.....

Reply:
Sounds good. I'll wait for #1765 to land, which adds `cached_property`